rust/src/http2/http2.rs

```diff
@@ -157,6 +157,8 @@ impl HTTP2Transaction {
     }
 
     pub fn free(&mut self) {
+        debug_validate_bug_on!(self.tx_data.files_opened > 1);
```

Comment:
Cannot we have 2 files ? 1 to client and 1 to server ?

Reply:
Yes you are correct. So if we have a way to only increment the file count when the file starts, we can use `incr_files_opened()` and check for max value of 2 here.

Reply:
It makes me wonder if the `ft: FileTransferTracker` in `HTTP2Transaction` should be duplicated as ft_ts and ft_tc

Reply:
Yes, I think so. HTTP1 uses this logic as well. A file container per direction.

Reply:
Can you implement this split in a separate PR? We'd want that to go into 6 as well. I'll then rework my branch on top of that.

Reply:
https://github.com/OISF/suricata/pull/6080